app/src/main/java/org/akvo/flow/util/SurveyIdGenerator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2017 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.akvo.flow.util;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class SurveyIdGenerator {
+
+    public SurveyIdGenerator() {
+    }
+
+    /**
+     * Get the survey id from the survey xml folder
+     * The structure can be either surveyId/survey.xml or surveyId/folder/survey.xml
+     *
+     * @param parts
+     * @return
+     */
+    @NonNull
+    public String getSurveyIdFromFilePath(@Nullable String[] parts) {
+        if (parts == null || parts.length <= 1) {
```

Comment:
For now, parts.lenght == 1 there is only the file name but when I make the change you requested, then this will also change.